samplecode/java/com/athenahealth/api/APIConnection.java

```diff
@@ -442,7 +438,7 @@ private Object call(String verb, String path, Map<String, String> parameters, Ma
 	        Object response;
 	        // We may have binary data coming back. Only use text-oriented
 	        // readers when the stream is not binary.
-	        if(null != contentType && contentType.startsWith("image/"))
+	        if(contentType != null && contentType.startsWith("image/"))
```

Comment:
I'm not an expert in Java, but the point of putting null first is to prevent accidental assignment (e.g. a typo of `if (null = contentType)` would be a compile error, but `if (contentType = null)` would result in unexpected, and wrong behavior)

Reply:
This is true in C, but not in Java. In Java, `contentType = null` returns `null` as the expression and the compiler will throw an error in this case. In short, in Java `if(contentType = null)` is not allowed and will throw compile time error. Refer this: https://stackoverflow.com/a/6883662/4747587

Reply:
I see